runtime/compiler/optimizer/InlinerTempForJ9.cpp

```diff
@@ -4780,6 +4780,18 @@ TR_J9InlinerPolicy::supressInliningRecognizedInitialCallee(TR_CallSite* callsite
          return true;
       case TR::java_lang_Class_cast:
          return true; // Call will be transformed into checkcast
+      case TR::java_lang_Object_hashCode:
+         if (callNode &&
+             callNode->getFirstChild() &&
+             callNode->getFirstChild()->getOpCode().hasSymbolReference())
+            {
+            TR::SymbolReference * classChildSymRef = callNode->getFirstChild()->getSymbolReference();
```

Comment:
I had a prior comment about whether this code assumes the call will be a direct one. If so, why is it okay to assume this ? I ask, because the code in the other commit gets the last child instead of the first one and that would skip over the vft load that would be the first child of an indirect call.

Reply:
Yes, this code was assuming the call will be a direct call, and this is based on the generated trees for all the methods except Object.hashCode. Object.hashCode gets transformed by VP from an indirect call to a direct call prior to the new hashcode folding transformation working on it. We also get the direct call version of Object.hashcode if the call node gets added in the root method from an inlined callee. However, at this stage of the compilation, since Object.hashCode call can be an indirect call, it was incorrect to use the first child here since it can be a vft load child. Using the last child is the correct approach to deal with that.